docs/sharing-data.md

```diff
@@ -23,10 +23,56 @@ add a **globals** block to your [Terramate configuration file](config-overview.m
 globals {
   env = "staging"
   values = [ 1, 2, 3 ]
+  obj = {
+    value = "string"
+  }
+}
+```
+
+The **globals** block also supports setting attributes inside an
+existent object. For the globals defined in the above example, the block
+below will set additional properties of the `global.obj` variable:
+
+```
+globals obj {
+  number = 10
+}
+```
+
+This will make the `global.obj` look like:
+
+```hcl
+{
+  value = "string"
+  number = 10
+}
+```
+
+In the same way, multiple labels can be provided, and Terramate will
+automatically create the object keys as needed to fulfill the user
+intent. Example:
+
+```
+globals "obj" "child" "grandchild" {
+  list = []
+}
+```
+
+This will make the `global.obj` look like:
+
+```hcl
+{
+  value = "string"
+  number = 10
+  child = {
+    grandchild = {
+      list = []
+    }
+  }
```

Comment:
maybe add something to make it clear that his extensions behavior also applies on the hierarchy, that globals inherited from the parent can be extended (disregard if I missed this info).